types/diff-match-patch/index.d.ts

```diff
@@ -1,63 +1,288 @@
 // Type definitions for diff-match-patch 1.0
 // Project: https://www.npmjs.com/package/diff-match-patch
 // Definitions by: Asana <https://asana.com>
+//                 ZHAO Jinxiang <https://github.com/xiaoxiangmoe>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.0
 
-export type Diff = [number, string];
+declare const enum Operation {
+    DELETE = -1,
+    INSERT = 1,
+    EQUAL = 0,
+}
+
+export type { Operation };
+
+export const DIFF_DELETE: Operation.DELETE;
+export const DIFF_INSERT: Operation.INSERT;
+export const DIFF_EQUAL: Operation.EQUAL;
 
+export type Diff = readonly [
+    /**
+     * Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL
+     */
+    operation: Operation,
+    /**
+     * Text to be deleted, inserted, or retained.
+     */
+    text: string
+];
```

Comment:
The enum notion does not exists in original code, can this be done just using aliases to provide proper types?
```ts
export const DIFF_DELETE: -1;
export const DIFF_INSERT: 1;
export const DIFF_EQUAL: 0;

export type Diff = readonly [
    /**
     * Operation, one of: DIFF_DELETE, DIFF_INSERT, DIFF_EQUAL
     */
    operation: typeof DIFF_DELETE | typeof DIFF_INSERT | typeof DIFF_EQUAL,
    /**
     * Text to be deleted, inserted, or retained.
     */
    text: string,
];
```

Reply:
But if we use `typeof DIFF_DELETE | typeof DIFF_INSERT | typeof DIFF_EQUAL` it will become `-1 | 0 | 1`